probes/releasesHaveProvenance/impl.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package releasesHaveProvenance
+
+import (
+	"embed"
+	"fmt"
+	"strings"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const (
+	Probe           = "releasesHaveProvenance"
+	releaseLookBack = 5
+)
+
+var provenanceExtensions = []string{".intoto.jsonl"}
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	var findings []finding.Finding
+
+	releases := raw.SignedReleasesResults.Releases
+
+	totalReleases := 0
+
+	for releaseIndex, release := range releases {
+		if len(release.Assets) == 0 {
+			continue
+		}
+		totalReleases++
+		hasProvenance := false
+		for assetIndex, asset := range release.Assets {
+			for _, suffix := range provenanceExtensions {
+				if !strings.HasSuffix(asset.Name, suffix) {
+					continue
+				}
+				// Create Positive Finding
+				// with file info
+				loc := &finding.Location{
+					Type: finding.FileTypeURL,
+					Path: asset.URL,
+				}
+				f, err := finding.NewWith(fs, Probe,
+					fmt.Sprintf("provenance for release artifact: %s", asset.Name),
+					loc,
+					finding.OutcomePositive)
+				if err != nil {
+					return nil, Probe, fmt.Errorf("create finding: %w", err)
+				}
+				f.Values = map[string]int{
+					"totalReleases": len(releases),
```

Comment:
Any particular reason to expose this data in `Finding.Values`? Not sure how relevant it is, other than `totalReleases`, which is used for score calculation. Would it be possible to expose release name/version instead?

Reply:
do we need this value at all? If there's one finding per release, it's implicit how many releases were found, no?

Reply:
> Would it be possible to expose release name/version instead?

I almost have this ready, so let's do it in this PR. Will get up in a bit.

Reply:
> If there's one finding per release, it's implicit how many releases were found, no?

There is one finding per release asset, i.e. one finding per artifact, key, provenance in each release.

Reply:
Done in https://github.com/ossf/scorecard/pull/3610/commits/8c65bb55cdb18ee18a806dcf7518e59bdf187661 with additional changes in https://github.com/ossf/scorecard/pull/3610/commits/d4feb833745bad65f8aee5814b83b2f728373092